### DISCUSSION TITLE
opus to PCM or Ogg format？

### DISCUSSION THREAD

**Discussion by zh383603842** [sipsorcery-org/sipsorcery#643] [General]
List<SDPAudioVideoMediaFormat> AudioOfferFormats = new List<SDPAudioVideoMediaFormat> {
new SDPAudioVideoMediaFormat(SDPMediaTypesEnum.audio, 111, "OPUS", 48000, 2, null)
};

peerConnection.OnRtpPacketReceived += (IPEndPoint rep, SDPMediaTypesEnum media, RTPPacket rtpPkt) =>
            {
                if (media == SDPMediaTypesEnum.audio)
                {
                    //opus to pcm?or ogg??
                }
            };

//How to convert opus to PCM or Ogg format？？？？

**Comment by sipsorcery** [comment]
There is no OPUS codec in any of the sipsorcery libraries. Likewise with the ogg continer format, There might be some .NET libraries floating around that implement or wrap an OPUS codec but you'd need to wire them iup yourself.

**Reply by zh383603842** [reply]
如何解码webrtc web传递过来的opus音频到pcm呢？我使用opusDecode无法正常解码是什么原因？

**Reply by sipsorcery** [reply]
https://github.com/DevJohnC/Opus.NET
